providers/provider_data.go

```diff
@@ -29,6 +29,7 @@ type ProviderData struct {
 	ProfileURL        *url.URL
 	ProtectedResource *url.URL
 	ValidateURL       *url.URL
+	GroupURL          *url.URL
```

Comment:
This isn't a common pattern for other providers, most of the time the groups should be included within the profile or ID tokens returned, do we really want to add this for all providers?